rmw_dds_common/include/rmw_dds_common/time_utils.hpp

```diff
@@ -0,0 +1,48 @@
+// Copyright 2020 Open Source Robotics Foundation, Inc.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+#ifndef RMW_DDS_COMMON__TIME_UTILS_HPP_
+#define RMW_DDS_COMMON__TIME_UTILS_HPP_
+
+#include "rmw/types.h"
+
+#include "rmw_dds_common/visibility_control.h"
+
+namespace rmw_dds_common
+{
+
+/// Check if an rmw_time_t's 64-bit seconds/nanoseconds values exceed
+/// 32-bit limits and adjust accordingly
+/*
+ * The DDS standard specifies the Time_t and Duration_t types with
+ * "long sec" and "unsigned long nanosec", and the IDL to C++11 mapping
+ * states that the C++ types for long and unsigned long are mapped
+ * to int32_t and uint32_t, respectively. So, the 64-bit seconds and
+ * nanoseconds values will be truncated to 32-bits when converted
+ * from rmw_time_t to either a DDS Duration_t or Time_t. This function
+ * limits the seconds value to the signed 32-bit maximum and attempts
+ * to move the excess seconds to the nanoseconds field. If the resulting
+ * nanoseconds value is too large for unsigned 32-bits it saturates
+ * and issues a warning.
+ * 
+ * \param[in] time to convert
+ * \return converted time value
+ */
+RMW_DDS_COMMON_PUBLIC
+rmw_time_t
+rmw_time_to_dds_time(const rmw_time_t & time);
```

Comment:
Now that we've changed the name to `rmw_time_to_dds_time`, the output parameter makes somewhat less sense to me.  That is, I would naively expect this function to take in an `rmw_time_t` (which it does), and return a DDS-RTPS `struct Time_t`.  Is it possible to do that, or do cross-vendor API considerations prevent us from doing that?

Reply:
I think it actually makes more sense to stick with `rmw_time_t` here: there are at least 3 different DDS API's in play here: the original ones that follow the IDL-to-language mapping (e.g., Connext's, Fast-RTPS's), a new C API (Cyclone), and the "new" C++11 DDS API (any new development is likely to use that interface). All 3 use different types to represent time stamps, and that makes `rmw_time_t` a good compromise.

Reply:
All right, that's fine by me.  At the risk of bikeshedding even more, then, I might suggest one more rename to something like:
`clamp_rmw_time_to_dds_time`.  @mjeronimo what do you think?

Reply:
Sure,that works for me. I'll update.